spring-cloud-contract-tools/spring-cloud-contract-converters/src/main/java/org/springframework/cloud/contract/verifier/wiremock/DefaultWireMockStubPostProcessor.java

```diff
@@ -43,7 +43,10 @@ public StubMapping postProcess(StubMapping stubMapping, Contract contract) {
 		StubMapping stubMappingFromMetadata = stubMappingFromMetadata(wireMockMetaData.getStubMapping());
 		stubMapping.setResponse(mergedResponse(stubMapping, stubMappingFromMetadata));
 		if (stubMappingFromMetadata.getPostServeActions() != null) {
-			setPostServeActions(stubMapping, stubMappingFromMetadata);
+			//FIXES ISSUE-1727
+			if(stubMapping.getPostServeActions() == null) {
```

Comment:
So what should happen if we already have post serve actions? Shouldn't we merge the two?

Reply:
From my observations the _postServeActions_ do not belong to _stubMapping_ by default, rather we are **injecting** them to the _stubMapping_ after we extract them from the contract at _postProcess_ stage. 

I see that’s a valid point about **merging** them, since the _postProcess_ gets invoked twice in this scenario. But _setPostServeActions()_ actually **appends** the _postServeActions_ to the existing _postServeActions_ of _stubMapping_ and does not merge them like _setMetadata_ does, since the underlying collection is **list** not a **map** which causes the issue.

 And from what I've seen the extracted _postServeActions_ from the _stubMappingFromMetadata_ are the same every time. Please correct me if I'm wrong. That's my first contribution to an open source project and I highly appreciate your feedback. :)

Reply:
> I see that’s a valid point about merging them, since the postProcess gets invoked twice in this scenario. But setPostServeActions() actually appends the postServeActions to the existing postServeActions of stubMapping and does not merge them like setMetadata does, since the underlying collection is list not a map which causes the issue.

why are we calling this twice?

> And from what I've seen the extracted postServeActions from the stubMappingFromMetadata are the same every time. Please correct me if I'm wrong. That's my first contribution to an open source project and I highly appreciate your feedback. :)

Ok, I'm starting to remember this... Indeed if they are already set they shouldn't be set again. The question remains why are we calling post process twice. You're doing great work with the contribution :)

Reply:
Great question. I've traced the execution flow very carefully and examined this:

1. convertASingleContract(1) calls postProcessStubMapping(2)
2. postProcessStubMapping(2) calls super.postProcessStubMapping(3)
3. super.postProcessStubMapping(3) calls the overriden defaultStubMappingPostProcessing(4) that's in the DslToWireMockClientConverter since processors are empty()
5. The overriden defaultStubMappingPostProcessing(4) calls postProcess(5) - which appends the postServeActions to the stub mapping - 1st time
6. After which we call defaultStubMappingPostProcessing is called once again at postProcessStubMapping - which repeats step 5 - thus appending postServeActions to stubMapping - 2nd time

The issue lies in the
```	@Override
	public StubMapping postProcessStubMapping(StubMapping stubMapping, Contract contract) {
		return super.postProcessStubMapping(stubMapping, contract); // that makes sure only 1 call is invoked in the post process stage
		// apply the default WireMock processor as the last one
		//return defaultStubMappingPostProcessing(mapping, contract); // removing this 
	}
```

I've commented out the last line that was calling the defaultStubMappingPostProcessing for the second time since it has been already executed in the super.postProcessStubMapping and removed the `if(stubMapping.getPostServeActions() == null)` that I've previously written.

Is there any reason you've called the super.postProcessStubMapping and then again defaultStubMappingPostProcessing in the first place? Is there anything that I'm breaking? I've ran all the tests and everything was passing. Thanks for the quick feedback :)

Reply:
> Is there any reason you've called the super.postProcessStubMapping and then again defaultStubMappingPostProcessing in the first place? Is there anything that I'm breaking? I've ran all the tests and everything was passing. Thanks for the quick feedback :)

Honestly? I don't remember ;) If the tests are passing that we could assume that things are working fine

Reply:
I've applied the changes we discussed. Will wait for your feedback once again. Thank you very much!

Reply:
It seems that my IDE deleted new line in one of the files and that's why build fails with checkstyle error. I'll make another commit.